src/Management/src/EndpointBase/Middleware/EndpointMiddleware.cs

```diff
@@ -8,87 +8,91 @@
 {
     protected IEndpoint<TResult> endpoint;
     protected ILogger logger;
     protected IManagementOptions managementOptions;
 
-    public EndpointMiddleware(IManagementOptions managementOptions, ILogger logger = null)
-    {
-        this.logger = logger;
-        this.managementOptions = managementOptions ?? throw new ArgumentNullException(nameof(managementOptions));
-        if (this.managementOptions is ManagementEndpointOptions options)
+        public EndpointMiddleware(IManagementOptions managementOptions, ILogger logger = null)
         {
-            options.SerializerOptions = GetSerializerOptions(options.SerializerOptions);
+            this.logger = logger;
+            this.managementOptions = managementOptions ?? throw new ArgumentNullException(nameof(managementOptions));
+            if (this.managementOptions is ManagementEndpointOptions options)
+            {
+                options.SerializerOptions = GetSerializerOptions(options.SerializerOptions);
+            }
         }
-    }
 
-    public EndpointMiddleware(IEndpoint<TResult> endpoint, IManagementOptions managementOptions, ILogger logger = null)
-        : this(managementOptions, logger)
-    {
-        this.endpoint = endpoint ?? throw new ArgumentNullException(nameof(endpoint));
-    }
+        public EndpointMiddleware(IEndpoint<TResult> endpoint, IManagementOptions managementOptions, ILogger logger = null)
+            : this(managementOptions, logger)
+        {
+            this.endpoint = endpoint ?? throw new ArgumentNullException(nameof(endpoint));
+        }
 
-    public IEndpoint<TResult> Endpoint
-    {
-        get => endpoint;
+        public IEndpoint<TResult> Endpoint
+        {
+            get => endpoint;
 
-        set => endpoint = value;
-    }
+            set => endpoint = value;
+        }
 
-    public virtual string HandleRequest()
-    {
-        var result = endpoint.Invoke();
-        return Serialize(result);
-    }
+        public virtual string HandleRequest()
+        {
+            var result = endpoint.Invoke();
+            return Serialize(result);
+        }
 
-    public virtual string Serialize(TResult result)
-    {
-        try
+        public virtual string Serialize(TResult result)
         {
-            JsonSerializerOptions options;
-            if (managementOptions is ManagementEndpointOptions endpointOptions)
+            try
             {
-                options = endpointOptions.SerializerOptions;
+                JsonSerializerOptions options;
+                if (managementOptions is ManagementEndpointOptions endpointOptions)
+                {
+                    options = endpointOptions.SerializerOptions;
+                }
+                else
+                {
+                    options = GetSerializerOptions(null);
+                }
+
+                return JsonSerializer.Serialize(result, options);
             }
-            else
+            catch (Exception e)
```

Comment:
Should we catch something more specific and potentially handle the [potentially thrown types](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.serialize?view=net-6.0) differently? Is there potential for `NotSupportedException` or do we definitely always know all the types being serialized?

Reply:
Good idea. Made tha change.